bot/exts/events/advent_of_code/_cog.py

```diff
@@ -172,6 +175,72 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
         else:
             await ctx.message.add_reaction(Emojis.envelope)
 
+    @in_month(Month.NOVEMBER, Month.DECEMBER)
+    @adventofcode_group.command(
+        name="link",
+        aliases=("connect",),
+        brief="Tie your Discord account with your Advent of Code name."
+    )
+    @whitelist_override(channels=AOC_WHITELIST)
+    async def aoc_link_account(self, ctx: commands.Context, aoc_name: str = None) -> None:
```

Comment:
why is aoc_name none here?

Reply:
The default is None in case the user just does `.aoc link` with no specified name. I can check the cache if they've already linked or now and then return the status. It doesn't need to error out if they don't remember linking or not.

Reply:
I see, that makes sense. Given the changes with #963, I feel like this command the leaderboard now have different input styles, which does not make the most sense.

Reply:
I'm not sure what you mean by this. All the user has to do is provide their advent of code leaderboard name and it gets linked, if they don't it'll tell them if they've linked it or not. It's the same as the `.aoc lb aoc_name` with that PR being merged.

Reply:
What #963 did is it made the user argument keyword only.

This means that a user can run `.aoc link name with spaces` and not need quotes. Given that this is what the bot now has and our users expect it to work (at least in the scope of the aoc commands), this command should not require quotes for a username argument.

This is not the same, as I've tested locally.
![image](https://user-images.githubusercontent.com/71233171/144651916-bb17975a-cbcb-47e1-a1c9-3f9faa33425b.png)

Current behavior treats it as a kw arg, but also strips the first set of `"` since that was the initial behavior. 
![image](https://user-images.githubusercontent.com/71233171/144652176-be952c8e-0276-4d02-b1f5-12e0cdf38fa0.png)

Reply:
Ah, okay. So your feedback is to make this keyword argument only to accept spaces in the name without quotes. I can do that. That wasn't clear from your original statement.